src/GeneratorV2/Process/Overloaders.cs

```diff
@@ -169,6 +169,229 @@ public void WritePrologue(IndentedTextWriter writer, NameTable nameTable)
         }
     }
 
+    public sealed class VectorOverloader : IOverloader
+    {
+        private static readonly Regex VectorNameMatch = new Regex("([1-4])([f|d|h|i])v$", RegexOptions.Compiled);
+        private static readonly Regex MatrixNameMatch = new Regex("Matrix([1-4])(?:x([1-4]))?([f|d])v$", RegexOptions.Compiled);
+        private readonly HashSet<string> _existingVectorTypes = new HashSet<string>()
+        {
+            "Vector2",
+            "Vector2d",
+            "Vector2h",
+            "Vector2i",
+            "Vector3",
+            "Vector3d",
+            "Vector3h",
+            "Vector3i",
+            "Vector4",
+            "Vector4d",
+            "Vector4h",
+            "Vector4i",
+
+            "Matrix2d",
+            "Matrix2x3d",
+            "Matrix2x4d",
+            "Matrix3x2d",
+            "Matrix3d",
+            "Matrix3x4d",
+            "Matrix4x2d",
+            "Matrix4x3d",
+            "Matrix4d",
+            "Matrix2",
+            "Matrix2x3",
+            "Matrix2x4",
+            "Matrix3x2",
+            "Matrix3",
+            "Matrix3x4",
+            "Matrix4x2",
+            "Matrix4x3",
+            "Matrix4",
+        };
+
+        public bool TryGenerateOverloads(Overload overload, [NotNullWhen(true)] out List<Overload>? newOverloads)
+        {
+            Match vectorMatch = VectorNameMatch.Match(overload.OverloadName);
+            Match matrixMatch = MatrixNameMatch.Match(overload.OverloadName);
+
+            int vectorSize;
+            string typePostfix;
+            string typeName;
+            if (matrixMatch.Success)
+            {
+                int columns = int.Parse(matrixMatch.Groups[1].Value);
+                int rows = columns;
+                typePostfix = matrixMatch.Groups[3].Value;
+                if (typePostfix == "f") typePostfix = "";
+                typeName = "Matrix" + columns;
+                if (matrixMatch.Groups[2].Success)
+                {
+                    rows = int.Parse(matrixMatch.Groups[2].Value);
+                    typeName += "x" + rows;
+                }
+                typeName += typePostfix;
```

Comment:
Here I would use string interpolation. And you don't even have to parse the numbers as all you want is to put them into a string. (requires regex change, see separate comment for that)
```cs
string dimentions = matrixMatch.Groups[1].Value;

typePostfix = matrixMatch.Groups[2].Value;
if (typePostfix == "f") typePostfix = "";

typeName = $"Matrix{dimentions}{typePostfix}";

vectorSize = dimentions[0] - '0'
if (dimentions.Length == 3)
    vectorSize *= dimentions[2] - '0';
```

Idk if this becomes any cleaner or more readable. It's just an alternative way of doing it.
The `dimentions[0] - '0'` might be particularly unintuitive so it might not be the best way.

Reply:
The problem with this is that we use the size later to check if the parameters len value has a different size. We dont wanna overload something with len="2" when we are looking for a Matrix4x4. I was looking for a way to extract both the numbers and the dimen*s*ions separately. But i couldn't make it work (note this is my first time working with regex.)

Reply:
I think what i changed this to is a lot simpler than what we both had beforehand.

Reply:
Yeah the new solution is great. Good work!